src/instructlab/process/process.py

```diff
@@ -0,0 +1,296 @@
+# Standard
+from datetime import datetime
+import json
+import logging
+import os
+import signal
+import subprocess
+import sys
+import time
+import uuid
+
+# Third Party
+import psutil
+
+# First Party
+from instructlab.configuration import DEFAULTS
+from instructlab.defaults import ILAB_PROCESS_MODES
+
+logger = logging.getLogger(__name__)
+
+
+def load_registry():
+    process_registry = {}
+    """Load the process registry from a file, if it exists."""
+    # we do not want a persistent registry in memory. This causes issues when in scenarios where you switch registry files (ex, in a unit test, or with multuple users)
+    # but the registry with incorrect processes still exists in memory.
+    if os.path.exists(DEFAULTS.REGISTRY_FILE):
+        with open(DEFAULTS.REGISTRY_FILE, "r") as f:
+            data = json.load(f)
+            for key, value in data.items():
+                process_registry[key] = value
+    else:
+        logger.info("No existing process registry found. Starting fresh.")
```

Comment:
We may want this to be a `debug` statement - not sure if normal users care about a process registry

Reply:
true, will switch